common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -49,6 +49,8 @@ case class MessageWithContext(message: String, context: java.util.HashMap[String
     resultMap.putAll(mdc.context)
     MessageWithContext(message + mdc.message, resultMap)
   }
+
+  override def toString: String = message
```

Comment:
Why override the method `toString`,

Dor the following scenarios, no need to write like as 
```
throw new RuntimeException(errMsg.toString + ....
```


```
val errMsg = log"checkCategoryMaps failed for input with " +
        log"maxCategories=${MDC(MAX_CATEGORIES, maxCategories)} " +
        log"categoricalFeatures=${MDC(CATEGORICAL_FEATURES, categoricalFeatures.mkString(", "))}"

...

throw new RuntimeException(errMsg + s". Categorical feature $feature failed" +
                  s" metadata check. Found feature attribute: $featureAttr.")
```